File: db4on/reference/vb/db4o-reference-chapters/src/com/db4odoc/f1/diagnostics/ArbitraryQuery.vb
' Copyright (C) 2004 - 2006 db4objects Inc. http://www.db4o.com 

Imports com.db4o.query

Namespace com.db4odoc.f1.diagnostics
    Public Class ArbitraryQuery
        Inherits Predicate
        Private _points() As Integer
        Public Sub New(ByVal points() As Integer)
            _points = points
        End Sub
        Public Function Match(ByVal pilot As evaluations.Pilot) As Boolean
            Dim points As Integer
            For Each points In _points
                If pilot.Points = points Then
                    Return True
                End If
            Next
            Return pilot.Name.StartsWith("Rubens")
        End Function
    End Class
End Namespace

File: db4on/reference/db4o-reference-chapters/src/com/db4odoc/f1/diagnostics/ArbitraryQuery.cs
/* Copyright (C) 2004 - 2006 db4objects Inc. http://www.db4o.com */

using com.db4o.query;

namespace com.db4odoc.f1.diagnostics
{
	public class ArbitraryQuery : Predicate
	{
		private int[] _points;
		public ArbitraryQuery(int[] points)
		{
			_points=points;
		}
		public bool Match(Pilot pilot)
		{
			foreach (int points in _points)
			{
				if (pilot.Points == points)
				{
					return true;
				}
			}
			return pilot.Name.StartsWith("Rubens");
		}
	}
}